src/logcollector/logcollector.c

```diff
@@ -1518,88 +1518,29 @@ int check_pattern_expand(int do_seek) {
                 break;
             }
 
-            char *global_path = NULL;
-            char *wildcard = NULL;
-
-            os_strdup(globs[j].gpath,global_path);
-
-            wildcard = strrchr(global_path,'\\');
+            char** result = expand_win32_wildcards(globs[j].gpath);
+            
+            if (result) {
 
-            if ( wildcard ) {
+                int file;
+                int totalFiles = 0;
+                char *full_path = NULL;
 
-                DIR *dir = NULL;
-                struct dirent *dirent = NULL;
-
-                *wildcard = '\0';
-                wildcard++;
-
-                if (dir = opendir(global_path), !dir) {
-                    merror("Couldn't open directory '%s' due to: %s", global_path, win_strerror(WSAGetLastError()));
-                    os_free(global_path);
-                    continue;
+                while (NULL != result[totalFiles])
+                {
+                    totalFiles++;
                 }
+                totalFiles %= maximum_files;
 
-                while (dirent = readdir(dir), dirent) {
-
-                    // Skip "." and ".."
-                    if (dirent->d_name[0] == '.' && (dirent->d_name[1] == '\0' || (dirent->d_name[1] == '.' && dirent->d_name[2] == '\0'))) {
-                        continue;
-                    }
-
-                    if (current_files >= maximum_files) {
-                        mwarn(FILE_LIMIT, maximum_files);
-                        break;
-                    }
```

Comment:
In order not to exceed the maximum number of configured files, we need this check inside the `for` loop.